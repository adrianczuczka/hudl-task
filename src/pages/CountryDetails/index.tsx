import React, { FC, useEffect, useState } from 'react';
import League from '../../network/models/league';
import { getLeaguesByCountry } from '../../network/networkService';
import AccordionCard from '../../components/AccordionCard';
import Accordion from 'react-bootstrap/Accordion';
import LoadingSpinner from '../../components/LoadingSpinner';
import Country from '../../network/models/country';
import LeagueDetails from '../LeagueDetails';

type Props = {
  country: Country;
  isOpen: boolean;
};

const CountryDetails: FC<Props> = ({ country, isOpen }: Props) => {
  const [leagues, setLeagues] = useState<League[]>([]);
  const [isLoading, setIsLoading] = useState(false);
  const [hasError, setHasError] = useState(false);
  const [selectedLeague, setSelectedLeague] = useState<League | undefined>(undefined);

  useEffect(() => {
    if (isOpen) {
      setIsLoading(true);
      getLeaguesByCountry(country.name).then((resp) => {
        setIsLoading(false);
        if (resp) {
          setLeagues(resp);
        } else {
          setHasError(true);
        }
      });
    }
  }, [isOpen]);

  const renderLeagues = () => {
    return leagues.map((league) => {
      return (
        <AccordionCard
          key={league.name}
          title={league.name}
          onClick={() => {
            setSelectedLeague(league);
          }}
          image={league.logo}
        >
          <LeagueDetails league={league} isOpen={selectedLeague?.ID === league.ID} />
        </AccordionCard>
      );
    });
  };

  if (isLoading) {
    return <LoadingSpinner centered />;
  }

  return <Accordion>{renderLeagues()}</Accordion>;
};

export default CountryDetails;
