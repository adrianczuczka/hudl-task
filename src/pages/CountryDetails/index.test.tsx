import React from 'react';
import CountryDetails from './index';
import Country from '../../network/models/country';
import renderer from 'react-test-renderer';
import League from '../../network/models/league';
import { act } from '@testing-library/react';

const mockLeagues = [new League({ ID: 1, name: 'test name', logo: 'test logo' })];

jest.mock('../../network/networkService', () => {
  return {
    getLeaguesByCountry: (country: string): Promise<League[] | null> => {
      return Promise.resolve(mockLeagues);
    },
  };
});

const country = new Country({ name: 'test name', code: 'test code', flag: 'test flag' });

it('builds tree', async () => {
  let tree;
  const component = <CountryDetails country={country} isOpen />;
  act(() => {
    tree = renderer.create(component).toJSON();
  });
  expect(tree).toMatchSnapshot();
});
