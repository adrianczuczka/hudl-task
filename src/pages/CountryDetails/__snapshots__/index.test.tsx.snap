// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`builds tree 1`] = `
<div
  className="accordion"
/>
`;
